x/ibc/20-transfer/keeper/relay.go

```diff
@@ -149,7 +150,14 @@ func (k Keeper) OnRecvPacket(ctx sdk.Context, packet channel.Packet, data types.
 	prefix := types.GetDenomPrefix(packet.GetDestPort(), packet.GetDestChannel())
 	source := strings.HasPrefix(data.Amount[0].Denom, prefix)
 
+	// decode the receiver address
+	receiver, err := sdk.AccAddressFromBech32(data.Receiver)
+	if err != nil {
+		return err
+	}
```

Comment:
@alessio's concern is valid. Here the transaction will fail if the address doesn't have the `cosmos` prefix. It won't pass because the counterparty chain won't have the prefix registered.

An alternative solution would be to split the address:

```go
prefix := strings.Split(data.Receiver, "1") // split right after the prefix
config: := sdk.GetConfig()
config.SetBech32PrefixForAccount(prefix[0], prefix[0]+sdk.PrefixPublic) // set the prefix to avoid errors

receiver, err := sdk.AccAddressFromBech32(data.Receiver)
if err != nil {
	return err
}

// reset to original chain prefix (this will also fail if the prefix of the chain is not "cosmos")
config.SetBech32PrefixForAccount(Bech32PrefixAccAddr, Bech32PrefixAccPub)
```

Longer term options:

1. Ideally, each chain should add a blacklist (or whitelist) of external chain prefixes that it doesn't (does) support on the config.
2. remove the error from `AccAddressFromBech32` altogether.

Reply:
Hmm, why? This is `onRecvPacket`, so the receiver address must be encoded with whatever prefix the receiving chain is using (otherwise an error should be thrown, as it is).